### DISCUSSION TITLE
Need guidance with Uno and Certificates

### DISCUSSION THREAD

**Discussion by axtens** [unoplatform/uno#14017] [Q&A]
Has anyone here had any experience with certificates and MSIX? I'm trying to get my head around the whole issue of how to distribute a Uno based MSIX and how to make running the app work. I'm not at all interested in the Microsoft Store. I started programming in 1977, so I'm having some cognitive dissonance, e.g. "Where's the EXE?"

The sideload/LOB approach is out. There's not much in the way of "Management".

How do certificates work when you've got a self-signed and how do they work when you've got a public? It seems that self-signed require all manner of user-intervention to permit whereas the public ones involve some but not as much. Would appreciate links to resources. The MS docs themselves I'm already aware of.

And being Uno, what impact the certificate on Mac/Linux/Mobile?

**Comment by jeromelaban** [comment]
Thanks for the questions!

> Has anyone here had any experience with certificates and MSIX? I'm trying to get my head around the whole issue of how to distribute a Uno based MSIX and how to make running the app work. I'm not at all interested in the Microsoft Store. I started programming in 1977, so I'm having some cognitive dissonance, e.g. "Where's the EXE?"

You can build WinAppSDK apps in unpackaged mode, which will https://learn.microsoft.com/en-us/windows/apps/windows-app-sdk/deploy-unpackaged-apps.  You can also use Uno's GTK or WPF heads which also produce standalone win32 based apps that you can run with a standard exe on Windows.

> How do certificates work when you've got a self-signed and how do they work when you've got a public? It seems that self-signed require all manner of user-intervention to permit whereas the public ones involve some but not as much. Would appreciate links to resources. The MS docs themselves I'm already aware of.

> And being Uno, what impact the certificate on Mac/Linux/Mobile?

This is not really needed if you use unpackaged apps with WinAppSDK, and there's no signing requirements for Uno's Gtk and WPF heads (when running on Linux and macOS).

For mobile targets, signing is required per store requirements, which you can https://learn.microsoft.com/en-us/xamarin/ios/deploy-test/app-distribution/app-store-distribution/publishing-to-the-app-store?tabs=windows.

**Comment by limeniye** [comment]
**Transferred from Upwork**

> Regarding certificates and MSIX, how does one distribute a Uno based MSIX without using Microsoft Store, that is, when side-loading? Does it work to side-load at all, or is the expectation of using MSIX that you will ultimately use the Store for distribution?
How do certificates work when self-signed and when public-signed? Self-signed appears to require all manner of user-intervention whereas public doesn't or does not as much. 
Being Uno, and thus "cross-platform", what impact, if any, does the certificate have on Mac/Linux/Mobile execution.
Provide links to resources/videos etc. The MS docs themselves I'm already aware of.

_The person was invited to the discussion._
